src/summaries.rs

```diff
@@ -85,15 +85,18 @@ pub struct Summary {
 /// Constructs a summary of a function body by processing state information gathered during
 /// abstract interpretation of the body.
 pub fn summarize(
-    environment: &Environment,
-    _inferred_preconditions: &List<AbstractValue>,
+    argument_count: usize,
+    exit_environment: &Environment,
     preconditions: &List<AbstractValue>,
     post_conditions: &List<AbstractValue>,
     unwind_condition: &List<AbstractValue>,
+    unwind_environment: &Environment,
 ) -> Summary {
-    let result = environment.value_at(&Path::LocalVariable { ordinal: 0 });
-    let side_effects: List<(Path, AbstractValue)> = List::new(); // todo: #31  extract from environment
-    let unwind_side_effects: List<(Path, AbstractValue)> = List::new(); // todo: #31  extract from environment
+    let result = exit_environment.value_at(&Path::LocalVariable { ordinal: 0 });
+    let side_effects: List<(Path, AbstractValue)> =
```

Comment:
I guess I missed this before, but why Lists instead of Vecs?

Reply:
Initially I thought that these things would be part of the per block environment, so having a functional data structure would increase sharing. As things turned out, however, a Vec will do nicely instead. I'll change it.